frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: position.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", newAssignment);
+            expect(resp).toHaveStatus("success");
+            assignment = resp.payload;
+
+            // Fetch the position related to other instructor
+            resp = await apiGET(`/admin/instructors`);
+            const otherInstructor = resp.payload.find(
+                (instructor) =>
+                    instructor.utorid ===
+                    databaseSeeder.seededData.instructors[4].utorid
+            );
+            expect(otherInstructor).toBeDefined();
+            resp = await apiGET(`/admin/sessions/${session.id}/positions`);
+            const otherPosition = resp.payload.find((position) =>
+                position.instructor_ids.find((id) => id === otherInstructor.id)
+            );
+            expect(otherPosition).toBeDefined();
+
+            // Create another assignment (with no DDAH)
+            // for position associated with other instructor
+            const otherAssignment = {
+                note: "",
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: otherPosition.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", otherAssignment);
+            expect(resp).toHaveStatus("success");
+            foreignAssignment = resp.payload;
+        });
 
-        // Get newly created instructor and create a DDAH for them
-        resp = await apiGET(`/admin/instructors`);
-        expect(resp).toHaveStatus("success");
-        const instructorId = resp.payload.find(
-            (instructor) => instructor.utorid === instructorUser.utorid
-        )?.id;
-        expect(instructorId).toBeDefined();
-        const newDDAH = await createDdahWithFixedDuties(instructorId);
+        beforeEach(async () => {
+            await switchToInstructorOnlyUser();
+        });
 
-        // Test the DDAH is fetched properly
-        await switchToInstructorOnlyUser();
-        resp = await apiGET(`/instructor/sessions/${session.id}/ddahs`);
-        expect(resp).toHaveStatus("success");
-        expect(resp.payload).toHaveLength(1);
-        expect(resp.payload[0]).toStrictEqual(newDDAH);
-    });
+        it("fetch Ddahs associated with self only", async () => {
+            // Our instructor should only be able to fetch ddahs associated with them
+            const resp = await apiGET(
+                `/instructor/sessions/${session.id}/ddahs`
+            );
+            expect(resp).toHaveStatus("success");
+
+            // Only one DDAH is intially seeded for this instructor
+            expect(resp.payload).toHaveLength(1);
```

Comment:
Even though this implementation should be fine, I think you should not assume that only one DDAH is initially seeded for the instructor unless you are seeding the instructor within the test. This is because someone may need to seed more DDAH for this particular instructor in the future and might run into problems with this test.

Reply:
While I don't like this approach either, that is similar to what's done [here](https://github.com/uoft-tapp/tapp/blob/fd163fd2b23401d63931250d1da9f6212634c416/frontend/src/tests/ddah-tests.js#L91) for the ddah tests. I see before it used `.toContain()` and I like that better too. I am not sure that's the direction we will be going in though. @siefkenj do you have a preference?

Reply:
I think you should fetch all DDAHs as admin and filter that list to contain only those for that instructor. Then check the list the instructor retrieves is the same.

Reply:
It's a great idea! For the implementation though, we don't have an endpoint to get all DDAHs per instructor, right? That means we will have to fetch all positions, filter them per instructor, then get all assignments for these positions and then get their respective DDAHs? That would complicate the test a lot imo, even though that will yield the most predictable outcome regardless of the seeded data.

That also conflicts with the idea that we "know the seeded data" and can rely on it unless I am missing something of course